### ISSUE TITLE
[Python] Implement new Concept Exercise: loops

### ISSUE THREAD

**Issue by DavidGerva** [exercism/v3#1430]
This issue describes how to implement the `loops` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This concept exercise should convey a basic understanding of how to handle `loops` in Python. This concept exercise should therefore teach the student the python `while` loops, `for` loops and should also teach how to interrupt or change the normal flow of a loop.

## Learning objectives

- use the `while` key to execute a loop: execute something as long as an expression is true
- use a `for` loop to iterate over an iterable object with the `for ... in` pattern. The Membership testing concept (`in`) is required to address this concept.
- understand the differences between `while` (condition-dependent) and `for` (length or iteration dependent) loops
- use the `break` statement to terminate the nearest enclosing loop
- use the `continue` statement to continue with the next cycle of the nearest enclosing loop
- use the `else` clause to perform operation when a `for` loop terminates

## Out of scope

- async loops (advanced)

## Concepts

- `while` loops
- `for` loops
- the `else` clause to perform operation when the loop terminates
- `break` statement
- `continue` statement

## Prerequisites

- iterable objects and iteration base knowledge
- collections
- the Membership testing concept (`in`)

## Resources to refer to

- [The while key](https://yawpitchroll.com/posts/the-35-words-you-need-to-python/#while)
- [The for loop](https://yawpitchroll.com/posts/the-35-words-you-need-to-python/#for)
- [The break statement](https://yawpitchroll.com/posts/the-35-words-you-need-to-python/#break)
- [The break statement python docs](https://docs.python.org/3/reference/simple_stmts.html#the-break-statement)
- [The continue statement](https://yawpitchroll.com/posts/the-35-words-you-need-to-python/#continue)
- [The continue statement python docs](https://docs.python.org/3/reference/simple_stmts.html#the-continue-statement)

### Hints

- TBD?

### After

Nothing

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named bool_basic_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
-

**Comment by aimorris** [comment]
This and #1432 should be merged to one concept exercise `loops`, where you can also explain the usage of `for` loops.

**Comment by DavidGerva** [comment]
> 
> 
> This and #1432 should be merged to one concept exercise `loops`, where you can also explain the usage of `for` loops.

I looked in the track/csharp, which I use as a guide, but I couldn't find the loop/while/for issues. They are two rather different loops that I would use in different logics, but just like in other languages, that's why I looked in the csharp. If you think they should merge, I will!
(the same applies for the other loop issues)

@cmccandless what do you think?

**Comment by cmccandless** [comment]
I think merging them makes sense; loops are a common concept among different languages, and since we're not targeting complete beginners here, it shouldn't be necessary to break that down further. In Python (as with most languages), any loop can be written with either `while` or `for`; the unique thing about Python is the `else` clause for loops. We will definitely want to highlight that in our exercise.

**Comment by DavidGerva** [comment]
Updates as per previous comments. These issues have been merged in this one: #1432 and #1394

**Comment by mohanrajanr** [comment]
Hey @BethanyG @DavidGerva : As I've already mentioned, since I am already working on the None Concept exercise, I would like to also take a stab at this. Let me know if that is okay?

**Comment by DavidGerva** [comment]
Sure! Thank you! And I should give a look at your other pr in the next hours.

**Comment by DavidGerva** [comment]
> Hey @BethanyG @DavidGerva : As I've already mentioned, since I am already working on the None Concept exercise, I would like to also take a stab at this. Let me know if that is okay?

Sure! Thank you! And I should give a look at your other pr in the next hours.

**Comment by mohanrajanr** [comment]
Created a PR for this Exercise here: https://github.com/exercism/v3/pull/2302

**Comment by BethanyG** [comment]
Since PR #2302 has been merged 🌟 , I am closing this issue.
